### PR TITLE
janitor: Update to Corrosion 0.4.0 release

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

**Comment by ogoffart** [comment]
CI is failling on Windows with

```
The following tests FAILED:
	  1 - test_datastructures (Exit code 0xc0000135)
	  2 - test_interpreter (Exit code 0xc0000135)
	  3 - test_properties (Exit code 0xc0000135)
	  4 - test_eventloop (Exit code 0xc0000135)
	  5 - test_models (Exit code 0xc0000135)
	  6 - test_window (Exit code 0xc0000135)
```

That exit code seems to be STATUS_DLL_NOT_FOUND, or 'A dependent dll was not found' (according to google search leading to stack overflow)

It worked with the v0.4-beta2 and the most likely commit is therefore https://github.com/corrosion-rs/corrosion/commit/de3f2f54c7a05984ed1fc7e0cddd91bb07a1b963

CC: @jschwe   any idea why our test can't find the dll with the release while it worked with the beta2?

**Comment by jschwe** [comment]
Would it be possible to run the CI job with Rust 1.69, just to confirm it's not related to rust 1.70? Rust 1.70 also added a new dependency on windows , which could also be related.

**Comment by jschwe** [comment]
Ah, I see you are setting `-DCMAKE_BUILD_TYPE=Debug` even for multi config generators.
I guess I'll have to extend https://github.com/corrosion-rs/corrosion/commit/de3f2f54c7a05984ed1fc7e0cddd91bb07a1b963 to explicitly check for multi-config generators and not rely on CMAKE_BUILD_TYPE being set implying a single config generator.

File: api/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ include(FetchContent)
 FetchContent_Declare(
     Corrosion
     GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
-    GIT_TAG v0.4-beta2
+    GIT_TAG v0.4.0
```
Comment (ogoffart) [inline api/cpp/CMakeLists.txt:14]:
```suggestion
    GIT_TAG v0.4.1
```

**Pull request merged**